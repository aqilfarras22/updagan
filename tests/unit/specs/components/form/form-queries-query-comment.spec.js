import { shallowMount, createLocalVue } from '@vue/test-utils';

import formQueriesQueryCommentComponent from '@/components/form/form-queries/form-queries-query-comment';
import formatGMTDate from '@/constants/filter';
import momentTimezone from 'moment-timezone';

// Vue.use(moment, {
//   momentTimezone,
// });

const { it, describe, expect } = global;

const testPropsData = {
  comment: {
    author: 'authorName',
    comment: 'comment',
    createdOn: 1,
  },
};

const localVue = createLocalVue();
localVue.use('moment', {
  momentTimezone,
});
localVue.use({
  install(vue) {
    vue.filter('moment', (v, format) => format);
    vue.filter(formatGMTDate);
  },
});

describe('components/form/form-queries/form-queries-query-comment.vue', () => {
  const createMountedWrapper = (options = {}) => shallowMount(formQueriesQueryCommentComponent, {
    localVue,
    propsData: { ...testPropsData },
    ...options,
  });
  it('should create instance', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('should able to receive props', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.vm.fakeProp).toBeUndefined();
    expect(wrapper.vm.comment).toEqual(testPropsData.comment);
  });

  it('has the expected html structure', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
