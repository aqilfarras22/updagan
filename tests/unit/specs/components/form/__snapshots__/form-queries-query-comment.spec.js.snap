// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/form-queries/form-queries-query-comment.vue has the expected html structure 1`] = `
<div
  class="comment-item"
>
  <div
    class="date"
  >
    <span>
      Jan 01, 1970 12:00 AM GMT
    </span>
  </div>
   
  <dl>
    <dt>
      <strong>
        authorName
      </strong>
    </dt>
     
    <dd>
      comment
    </dd>
  </dl>
</div>
`;
